components/google_calendar/google_calendar.app.js

```diff
@@ -1,154 +1,307 @@
-    updatedMin: { description: "Lower bound for an event's last modification time (as a RFC3339 timestamp) to filter by. When specified, entries deleted since this time will always be included regardless of showDeleted. Optional. The default is not to filter by last modification time.",
+    role: {
+      label: "Role",
+      type: "string",
+      description: "The role assigned to the scope.",
       optional: true,
-      type: "string"
-    }
+      options() {
+        const roles = [
+          "none",
+          "freeBusyReader",
+          "reader",
+          "writer",
+          "owner",
+        ];
+        const options = roles.map((role) => {
+          return {
+            label: role,
+            value: role,
+          };
+        });
+        return options;
+      },
+    },
+    scopeType: {
+      label: "Type of the Scope",
+      type: "string",
+      description: "The extent to which calendar access is granted by this ACL rule.",
+      optional: true,
+      default: "default",
+      options() {
+        const types = [
+          "user",
+          "group",
+          "domain",
+        ];
+        const options = types.map((type) => {
+          return {
+            label: type,
+            value: type,
+          };
+        });
+        return options;
+      },
+    },
+    scopeValue: {
+      label: "Type of the Scope",
+      type: "string",
+      description: "The email address of a user or group, or the name of a domain, depending on the scope type. Omitted for type 'default'",
+      optional: true,
+      default: "",
+    },
   },
   methods: {
     _tokens() {
-      const access_token = get(this, "$auth.oauth_access_token")
-      const refresh_token = get(this, "$auth.oauth_refresh_token")
+      const access_token = get(this, "$auth.oauth_access_token");
+      const refresh_token = get(this, "$auth.oauth_refresh_token");
       return {
         access_token,
         refresh_token,
-      }
+      };
     },
     // returns a calendar object you can do whatever you want with
     calendar() {
-      const auth = new google.auth.OAuth2()
-      auth.setCredentials(this._tokens())
-      const calendar = google.calendar({version: "v3", auth})
-      return calendar
+      const auth = new google.auth.OAuth2();
+      auth.setCredentials(this._tokens());
+      const calendar = google.calendar({
+        version: "v3",
+        auth,
+      });
+      return calendar;
     },
     async calendarList() {
-      const calendar = this.calendar()
-      const resp = await calendar.calendarList.list()
-      return resp
+      const calendar = this.calendar();
+      const resp = await calendar.calendarList.list();
+      return resp;
     },
     async list(config) {
-      const calendar = this.calendar()
-      const resp = await calendar.events.list(config)
-      return resp
+      const calendar = this.calendar();
+      const resp = await calendar.events.list(config);
+      return resp;
     },
```

Comment:
I would suggest refactoring this function `list` to:
```javascript
    async listEvents(params) {
      const calendar = this.calendar();
      const { data } = await calendar.events.list(params);
      return data;
    },
```

In that way you can remove the function `getEvents` having in mind that you will then need to modify source components that reference that function changing it to `listEvents`